### PR TITLE
In File > New Dataframe, added the Sequence data type to the new dataframe grid

### PR DESCRIPTION
**Pull request by N-thony**
Fixes (partially) #7387 
@rdstern I have added the Sequence data type as suggested in the corresponding issue, please take a look.
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

**Review by rdstern — DISMISSED** [review]
@N-thony great - this works fine for me!  Over to @lloyddewit for possible merging.  The other part is less urgent, and can come later.

**Comment by lloyddewit** [comment]
@N-thony may I review or are you still updating? thanks

**Comment by N-thony** [comment]
> @N-thony may I review or are you still updating? thanks

You can review. Thanks.

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony If you are going to do it they way you have suggested, then it has got to be cleverer code than this, see below:
image

The regular sequence dialogue works fine for this example.
image

Good to have those examples,  After NA have 0, then 1, then abc, then 1,10 etc.

File: instat/dlgNewDataFrame.vb [hunk 1]
```diff
@@ -353,21 +354,33 @@ Public Class dlgNewDataFrame
...
 
-            Dim strColumnName As String = row.Cells("colNames").Value
-            clsColExpRFunction.AddParameter("x", clsRFunctionParameter:=clsEmptyRepFunction, bIncludeArgumentName:=False, iPosition:=0)
-            clsNewDataFrameFunction.AddParameter(strColumnName, clsRFunctionParameter:=clsColExpRFunction, iPosition:=iColPosition)
-            iColPosition += 1
+                Dim strColumnName As String = row.Cells("colNames").Value
+                clsColExpRFunction.ClearParameters()
+                If (strType = "Integer" OrElse strType = "Numeric") AndAlso Not strDefault = "NA" _
+                                        AndAlso IsNumeric(strDefault) AndAlso strDefault.Contains(",") Then
+                    Dim iTemp As Integer = iRows + CInt(strDefault) - 1
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:373]:
```suggestion
```

File: instat/dlgNewDataFrame.vb [hunk 2]
```diff
@@ -480,8 +492,32 @@ Public Class dlgNewDataFrame
                 dataTypeGridView(iColumnIndex, iRowIndex).Value = e.FormattedValue
             End If
         End If
+        If e.ColumnIndex = 3 Then
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:495]:
```suggestion
        ElseIf e.ColumnIndex = 3 Then
```

File: instat/dlgNewDataFrame.vb [hunk 3]
```diff
@@ -480,8 +492,32 @@ Public Class dlgNewDataFrame
...
+    'If Not Integer.TryParse(e.Value.ToString(), Nothing) Then
+    '    cell.ErrorText = $"Invalid Value: {e.Value}"
+    '    e.Value = cell.Value
+    '    e.ParsingApplied = True
+
+    'Else
+    '    cell.ErrorText = String.Empty
+    'End If
+    'End Sub
+
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:520]:
```suggestion
```

File: instat/dlgNewDataFrame.vb [hunk 4]
```diff
@@ -507,23 +543,33 @@ Public Class dlgNewDataFrame
...
+                dataTypeGridView(iColumnLevelIndex, iRowLevelIndex).ReadOnly = True
+                dataTypeGridView(iColumnLevelIndex, iRowLevelIndex).Value = ""
+
             End If
+            'If selectedCombobox.SelectedItem = "Integer" Then
+            '    'dataTypeGridView(iColumnDefaultIndex, iRowDefaultIndex).ValueType = GetType(Integer)
+            '    dataTypeGridView(iColumnDefaultIndex, iRowDefaultIndex).Value = 1
+            'Else
+            '    'dataTypeGridView(iColumnDefaultIndex, iRowDefaultIndex).ValueType = GetType(String)
+            'End If
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:572]:
```suggestion
```

File: instat/dlgNewDataFrame.vb [hunk 5]
```diff
@@ -507,23 +543,33 @@ Public Class dlgNewDataFrame
...
             Dim selectedCombobox As ComboBox = DirectCast(sender, ComboBox)
             If selectedCombobox.SelectedItem = "Factor" Then
-                dataTypeGridView(iColumnIndex, iRowIndex).ReadOnly = False
+                dataTypeGridView(iColumnLevelIndex, iRowLevelIndex).ReadOnly = False
             Else
-                dataTypeGridView(iColumnIndex, iRowIndex).ReadOnly = True
-                dataTypeGridView(iColumnIndex, iRowIndex).Value = ""
+                dataTypeGridView(iColumnLevelIndex, iRowLevelIndex).ReadOnly = True
+                dataTypeGridView(iColumnLevelIndex, iRowLevelIndex).Value = ""
+
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:565]:
```suggestion
```

File: instat/dlgNewDataFrame.vb [hunk 6]
```diff
@@ -507,23 +543,33 @@ Public Class dlgNewDataFrame
...
     End Sub
 
     Private Sub selectedComboBox_SelectionChangeCommitted(ByVal sender As Object, ByVal e As EventArgs)
         If dataTypeGridView.CurrentCell.ColumnIndex = 2 Then
-            Dim iColumnIndex As Integer = dataTypeGridView.CurrentRow.Cells("colLevels").ColumnIndex
-            Dim iRowIndex As Integer = dataTypeGridView.CurrentRow.Cells("colLevels").RowIndex
+            Dim iColumnLevelIndex As Integer = dataTypeGridView.CurrentRow.Cells("colLevels").ColumnIndex
+            Dim iRowLevelIndex As Integer = dataTypeGridView.CurrentRow.Cells("colLevels").RowIndex
+            Dim iColumnDefaultIndex As Integer = dataTypeGridView.CurrentRow.Cells("colDefault").ColumnIndex
+            Dim iRowDefaultIndex As Integer = dataTypeGridView.CurrentRow.Cells("colDefault").RowIndex
```
Comment (lloyddewit) [inline instat/dlgNewDataFrame.vb:558]:
```suggestion
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
@rdstern could you test this now? Thanks.

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony the sequence looks fine now.
However, the factor type doesn't seem to work now

**Comment by N-thony** [comment]
> @N-thony the sequence looks fine now. However, the factor type doesn't seem to work now

@rdstern this is fixed now.

File: instat/dlgNewDataFrame.vb (hunk 1 above)
Comment (lloyddewit) [reply instat/dlgNewDataFrame.vb:375]: still needs resolving

**Comment by lloyddewit** [comment]
@N-thony from my previous review, there's still one comment that needs resolving
thanks

**Comment by lloyddewit** [comment]
I will review the recent changes

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
> @N-thony from my previous review, there's still one comment that needs resolving thanks

@N-thony Please ignore, I made the change, I think its uncontroversial.
Thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**